src/main/webapp/js/common.js

```diff
@@ -870,3 +870,64 @@ function hideSingleCollapse(e) {
     $(heading).find('a.btn').hide();
 }
 
+/**
+ * Wrapper for Bootbox.js (available at http://bootboxjs.com/)
+ * "Bootbox.js is a small JavaScript library which allows you to create programmatic dialog boxes using
+ *  Bootstrap modals"
+ */
+var BootboxWrapper = {
+
+    /**
+     * Custom alert dialog to replace default alert() function
+     * Required params: titleText and messegeText
+     * Optional params: okButtonText (defaults to "OK")
+     *                  color (defaults to StatusType.INFO)
+     */
+    showModalAlert: function(titleText, messageText, okButtonText, color) {
+        bootbox.dialog({
+            title: titleText,
+            message: messageText,
+            buttons: {
+                okay: {
+                    label: okButtonText || 'OK',
+                    className: 'modal-btn-ok '
+                               + 'btn-' + color || StatusType.INFO
+                }
+            }
+        })
+        // applies bootstrap color to title background
+        .find('.modal-header').addClass('alert-' + color || StatusType.INFO);
+    },
+
+    /**
+     * Custom confirmation dialog to replace default confirm() function
+     * Required params: titleText, messegeText and okCallback
+     * Optional params: cancelCallBack (defaults to null)
+     *                  okButtonText (defaults to "OK")
+     *                  cancelButtonText (defaults to "Cancel")
+     *                  color (defaults to StatusType.INFO)
+     */
+    showModalConfirmation: function(titleText, messageText, okCallback, cancelCallback,
+                                    okButtonText, cancelButtonText, color) {
+        bootbox.dialog({
+            title: titleText,
+            message: messageText,
+            buttons: {
+                cancel: {
+                    label: cancelButtonText || 'Cancel',
+                    className: 'modal-btn-cancel '
+                               + 'btn-default',
+                    callback: cancelCallback || null
+                },
+                ok: {
+                    label: okButtonText || 'OK',
```

Comment:
@wkurniawan07 I think using constants for default settings is okay (actually it's good).

I just don't want the constants to be in global scope because it suggests that the constants should be used and thus encourages the _passing in of those constants_.

In fact, the constants should not be used in normal operations because they represents default settings, and there shouldn't be a need to set the default settings by passing them into the method.

I guess I'm just not happy with the current wrapper method because the user cannot just set the color and use the other default values without passing in null (or the default values themselves).

Maybe we can pass in a JS object (like a dict/hash) into the wrapper method